Allow to disable Javascript

roman-mibex-2 opened the pull request:
In an environment where you run
untrusted Jupyter notebooks,
running the JavaScript is dangerous.
It opens the path to XSS attacks.

Therefore, allow to disable JavaScript
execution by allowing to provide a function
which renders the JavaScript. Then an alternative
function can be supplied which doesn't execute the script

Commit:
Allow to disable Javascript

In an environment where you run
untrusted Jupyter notebooks,
running the JavaScript is dangerous.
It opens the path to XSS attacks.

Therefore, allow to disable JavaScript
execution by allowing to provide a function
which renders the JavaScript. Then an alternative
function can be supplied which doesn't execute the script

roman-mibex-2 commented:
Thinks to discuss / consider.


- [ ] Maybe it just should be a boolean flag to enable/disable JavaScript execution?


- [ ] I personally would switch the default: Do not execute JavaScript by default! Opt in for JavaScript execution.
  If someone uses the library (like we did), they are maybe not aware that they the library supports arbitrary JavaScript execution. So, I would disable that 'feature' by default and make it opt in.
  Of course, switching the default will break existing users. But maybe also protect them from XSS their not aware of.
  

I'm happy to do the changes / update the pull request.

Commit:
Use Browsers APIs in Readme example

Use well known Browser APIs in README
examples, instead of internal functions.

jsvine commented:
Hi @roman-mibex-2, and thank you. I like your proposal. Notes:

- I agree that this should be a boolean, at least for now. For that reason, I'd suggest renaming the option to `executeJavaScript`
- I also agree that this option should default to `false`

If that sounds good to you, you can push those changes and then I will review again before merging.

Commit:
Do Not Run JavaScript By Default

Change the default to no run JavaScript by default.
Provide the `executeJavaScript` to enable executing
JavaScript in Notebooks.

roman-mibex-2 commented:
@jsvine Updated the PR to use the executeJavaScript, with disabled JavaScript by default.
